src/components/videoScreen/VideoScreen.tsx

```diff
@@ -0,0 +1,130 @@
+import s from "./videoScreen.module.scss";
+import MicOff from "../../assets/MicOff.svg";
+import MicOn from "../../assets/MicOn.svg";
+import { useDispatch, useSelector } from "react-redux";
+import {
+  selectCameraOn,
+  selectMicoOn,
+  selectUser,
+  selectBgOn,
+} from "../../redux/selectors";
+import { setBasicSettings } from "../../redux/settingsSlice";
+import MicBadge from "../micBadge/MicBadge";
+import CameraBadge from "../cameraBadge/CameraBadge";
+import BackgroundBadge from "../backgroundBadge/BackgroundBadge";
+import SettingsBadge from "../settingsBadge/SettingsBadge";
+import NameComponent from "../nameComponent/NameComponent";
+import { useRef, useState } from "react";
+
+function VideoScreen() {
+  const [error, setError] = useState<null | string>(null);
+  const [micStream, setMicStream] = useState<MediaStream | null>(null);
+  const [cameraStream, setCameraStream] = useState<MediaStream | null>(null);
+  const videoRef = useRef<HTMLVideoElement | null>(null);
+  const dispatch = useDispatch();
+  const isMicOn = useSelector(selectMicoOn);
+  const isCameraOn = useSelector(selectCameraOn);
+  const isBgSelect = useSelector(selectBgOn);
+  const name = useSelector(selectUser);
+
+  const handleMicClick = async () => {
+    if (!isMicOn) {
```

Comment:
Use ! + return, do not create unnecessary nesting

Reply:
fixed